Dockerfile

```diff
@@ -31,35 +31,51 @@ RUN pip install --target=/dependencies playwright~=1.27.1 \
 # Final image stage
 FROM python:3.10-slim
 
-RUN apt-get update && apt-get install -y --no-install-recommends \
-    libssl1.1 \
-    libxslt1.1 \
-    # For pdftohtml
-    poppler-utils \
-    zlib1g \
-    && apt-get clean && rm -rf /var/lib/apt/lists/*
-
-
-# https://stackoverflow.com/questions/58701233/docker-logs-erroneously-appears-empty-until-container-stops
-ENV PYTHONUNBUFFERED=1
 
-RUN [ ! -d "/datastore" ] && mkdir /datastore
+RUN set -ex; \
+    apt-get update && apt-get install -y --no-install-recommends \
+        gosu \
+        libssl1.1 \
+        libxslt1.1 \
+        # For pdftohtml
+        poppler-utils \
+        zlib1g && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*; \
+    useradd -u 911 -U -M -s /bin/false changedetection && \
+    usermod -G users changedetection; \
+    mkdir -p /datastore
```

Comment:
mkdir if not exists? will it work if it exists?

Reply:
`mkdir -p` prevents the command to fail upon existing directory, so the exit code will be 0.